### PR TITLE
feat: add Prometheus service discovery in Kubernetes

### PR DESCRIPTION
**Pull request by gxthrj**
- [x] 0.Add Prometheus service discovery in kubernetes 
- [x] 1.fix: Add string type matching in overrideSlice
- [x] 2.fix: Initialize runningQueue in FetcherGatherer
- [x] 3.fix: Initialize OutputChannel in Prometheus fetcher plugin

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please fix CI.

File: configs/satellite_config.yaml [hunk 1]
```diff
@@ -247,3 +247,51 @@ pipes:
       client_name: grpc-client
       forwarders:
         - plugin_name: nativemeter-grpc-forwarder
+  - common_config:
```
Comment (wu-sheng) [inline configs/satellite_config.yaml:250]: Should the fetcher pipeline be in the document only?
I think in default, we don't scratch from anywhere.

Comment (gxthrj) [reply configs/satellite_config.yaml:250]: OK, will move to docs

File: configs/satellite_config.yaml [hunk 2]
```diff
@@ -247,3 +247,51 @@ pipes:
...
+        plugin_name: "memory-queue"
+        # The maximum buffer event size.
+        event_buffer_size: ${SATELLITE_QUEUE_EVENT_BUFFER_SIZE:5000}
+    processor:
+      filters:
+    sender:
+      fallbacker:
+        plugin_name: none-fallbacker
+      # The time interval between two flush operations. And the time unit is millisecond.
+      flush_time: ${SATELLITE_LOGPIPE_SENDER_FLUSH_TIME:1000}
```
Comment (mrproliu) [inline configs/satellite_config.yaml:290]: Please rename the environment key, is not the log pipeline.

**Review by mrproliu — COMMENTED** [review]
Basic LGTM.

File: configs/satellite_config.yaml (hunk 2 above)
Comment (gxthrj) [reply configs/satellite_config.yaml]: instead with `SATELLITE_PROMETHEUS_*`

**Review by mrproliu — APPROVED** [review]

**Comment by wu-sheng** [comment]
@gxthrj Do you add e2e test in this PR or in a separate one? We at least need it in this week, so we could include it in 0.2.0 release.

**Comment by wu-sheng** [comment]
Changelog should be updated, https://github.com/apache/skywalking-satellite/blob/main/CHANGES.md. 

There is a format issue for bug section, please fix it too.

**Comment by gxthrj** [comment]
> @gxthrj Do you add e2e test in this PR or in a separate one? We at least need it in this week, so we could include it in 0.2.0 release.

Will add e2e test in a separate one, and try my best to make it done in this week.

**Comment by gxthrj** [comment]
Fix https://github.com/apache/skywalking/issues/7436

**Pull request merged**